standalone-integration/src/test/java/com/ericsson/bss/cassandra/ecchronos/standalone/ITOnDemandRepairJob.java

```diff
@@ -0,0 +1,317 @@
+    }
+
+    @After
+    public void clean()
+    {
+        for (String keyspace : myCreatedKeyspaces)
+        {
+            for (TableMetadata tableMetadata : myMetadata.getKeyspace(keyspace).getTables())
+            {
+                mySession.execute(QueryBuilder.delete()
+                        .from("system_distributed", "repair_history")
+                        .where(QueryBuilder.eq("keyspace_name", keyspace))
+                        .and(QueryBuilder.eq("columnfamily_name", tableMetadata.getName())));
+            }
+
+            mySession.execute(SchemaBuilder.dropKeyspace(keyspace).ifExists());
+        }
+        myHostStates.close();
+        myRepairSchedulerImpl.close();
+        myScheduleManagerImpl.close();
+        myLockFactory.close();
+    }
+
+    /**
+     * Create a table that is replicated and schedule a repair on it
+     */
+    @Test
+    public void repairSingleTable() throws EcChronosException
+    {
+        long startTime = System.currentTimeMillis();
+        TableReference tableReference = new TableReference("ks", "tb");
+
+        createKeyspace(tableReference.getKeyspace(), 3);
+        createTable(tableReference);
+
+        myRepairSchedulerImpl.scheduleJob(tableReference);
+
+        await().pollInterval(1, TimeUnit.SECONDS).atMost(90, TimeUnit.SECONDS)
+                .until(() -> isRepairedSince(tableReference, startTime));
+
+        verifyTableRepairedSince(tableReference, startTime);
+    }
+
+    /**
+     * Create two tables that are replicated and repair both
+     */
+    @Test
+    public void repairMultipleTables() throws EcChronosException
+    {
+        long startTime = System.currentTimeMillis();
+
+        TableReference tableReference = new TableReference("ks", "tb");
+        TableReference tableReference2 = new TableReference("ks", "tb2");
+
+        createKeyspace(tableReference.getKeyspace(), 3);
+        createTable(tableReference);
+        createTable(tableReference2);
+
+        myRepairSchedulerImpl.scheduleJob(tableReference);
+        myRepairSchedulerImpl.scheduleJob(tableReference2);
+
+        await().pollInterval(1, TimeUnit.SECONDS).atMost(90, TimeUnit.SECONDS)
+                .until(() -> myRepairSchedulerImpl.getCurrentRepairJobs().isEmpty());
+
+        verifyTableRepairedSince(tableReference, startTime);
+        verifyTableRepairedSince(tableReference2, startTime);
+    }
+
+    /**
+     * Schedule two jobs on the same table
+     */
+    @Test
+    public void repairSameTableTwice() throws EcChronosException
+    {
+        long startTime = System.currentTimeMillis();
+
+        TableReference tableReference = new TableReference("ks", "tb");
+
+        createKeyspace(tableReference.getKeyspace(), 3);
+        createTable(tableReference);
+
+        myRepairSchedulerImpl.scheduleJob(tableReference);
+        myRepairSchedulerImpl.scheduleJob(tableReference);
+        await().pollInterval(1, TimeUnit.SECONDS).atMost(90, TimeUnit.SECONDS)
```

Comment:
There currently seems to be three different await() checks for the different test cases.

Would it make sense to simply wait for `myRepairSchedulerImpl.getCurrentRepairJobs().isEmpty()` in all test cases and then perform additional asserts?